corehq/apps/app_manager/static/app_manager/js/modules/module_view.js

```diff
@@ -56,6 +56,14 @@ hqDefine("app_manager/js/modules/module_view", function () {
         }
 
         var originalCaseType = initial_page_data('case_type');
+        var casesExist = false;
+        $.ajax({
+            method: 'GET',
+            url: hqImport('hqwebapp/js/initial_page_data').reverse('existing_case_types'),
+            success: function (data) {
+                casesExist = data.existing_case_types.includes(originalCaseType);
```

Comment:
The warning may not show even when cases do exist since this is set in an async call and the network can fail or be delayed longer than it takes for a user to interact with this page. Does that seem like it could be a problem?

Would it make sense make `casesExist` default to `true` so the warning is shown even if the ajax call fails? One problem with this approach is that the warning will show for a short amount of time until the ajax call succeeds. Solving that seems tricky. Maybe a reasonable compromise is to put an error handler on the ajax call and unconditionally set `casesExist = true` if the call fails? That still does not solve the issue of page interactions occurring faster than the ajax call response time.

Reply:
Another alternative: disable the input and maybe put a (small, unobstrusive) message below saying something vague like "Checking server for data..."

Yet another alternative: decide it doesn't matter, that the worst consequence (not showing the warning) isn't worth the extra clutter / mental overhead for users introduced by error display.

In any case, I do agree that it'd be good to explicitly decide how to handle errors and delays.

Reply:
Dev and I agree that we should not show the warning until the async call successfully returns.  I've added a comment to explain this behavior.